Mathlib/Algebra/Polynomial/Degree/Domain.lean

```diff
@@ -40,13 +40,15 @@ lemma natDegree_mul (hp : p ≠ 0) (hq : q ≠ 0) : (p*q).natDegree = p.natDegre
   rw [← Nat.cast_inj (R := WithBot ℕ), ← degree_eq_natDegree (mul_ne_zero hp hq),
     Nat.cast_add, ← degree_eq_natDegree hp, ← degree_eq_natDegree hq, degree_mul]
 
+omit [NoZeroDivisors R] in
 variable (p) in
-lemma natDegree_smul (ha : a ≠ 0) : (a • p).natDegree = p.natDegree := by
+lemma natDegree_smul {R' : Type*} [Zero R'] [SMulWithZero R' R] [NoZeroSMulDivisors R' R] {a : R'}
+    (ha : a ≠ 0) : (a • p).natDegree = p.natDegree := by
```

Comment:
For a future PR; does this generalize further to `IsSMulRegular a R`, allowing you to drop the zero requirement? That would let this apply to group actions.

Reply:
> does this generalize further to `IsSMulRegular a R`, allowing you to drop the zero requirement?

It seems that currently `SMul _ R[X]` are synthesized from https://github.com/leanprover-community/mathlib4/blob/f6b3bf487bff9b227d4435ac186541716ea54978/Mathlib/Algebra/Polynomial/Basic.lean#L136-L138, so it may need new instance defined to allow to drop `SMulZeroClass`?

Reply:
I'm suggesting that you can weaken `SMulWithZero` to `SMulZeroClass`

Reply:
Oops, I see. I was confused by their names a bit.